src/libraries/System.Globalization/tests/CompareInfo/CompareInfoTests.cs

```diff
@@ -359,6 +360,14 @@ public void SortKeyKanaTest(CompareInfo compareInfo, string string1, string stri
             SortKeyTest(compareInfo, string1, string2, options, expected);
         }
 
+
+        [DllImport("kernel32", CharSet = CharSet.Unicode)]
+        private static extern int CompareStringEx(string lpLocaleName, uint dwCmpFlags, string lpString1, int cchCount1, string lpString2, int cchCount2, IntPtr lpVersionInformation, IntPtr lpReserved, int lParam);
+        private const int NORM_LINGUISTIC_CASING = 0x08000000;       // use linguistic rules for casing
+
+        private static bool WindowsVersionHasTheCompareStringRegression =>
```

Comment:
What does "the compare string regression" mean?  Will we understand which regression it's referring to in a month or a year?

Reply:
The code is clear I guess. comparing 2 specific strings returning wrong results. I expect we'll remove this code when Windows fix the regression too. I added this code as I don't know specific versions will have the regression and when it is going to be fixed.

Reply:
@stephentoub please let me know if you have any concern before I merge this one.

Reply:
Might be worth a comment, but otherwise, ok.